### PR TITLE
utf8cpp: init at 3.1.1

### PR DESCRIPTION
**Pull request by jobojeha**
###### Motivation for this change
Adds UTF8-CPP, a  simple, portable and lightweight generic library for handling UTF-8 encoded strings. Required as a dependency for ParaView > 5.7.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -3646,6 +3646,12 @@
     githubId = 1102396;
     name = "Jussi Maki";
   };
+  jobojeha = {
```
Comment (IvarWithoutBones) [inline maintainers/maintainer-list.nix:3649]: This should be in a seperate commit

File: pkgs/development/libraries/utf8cpp/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    sha256 = "1d5p9l27mxhsmm4zlxjsbxngsmr5yxrzi66x550bf5g92sj52ffc";
+  };
+
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=None"
+    "-DCMAKE_INSTALL_LIBDIR=lib"
+    "-DUTF8_TESTS=OFF"
+  ];
+
+  enableParallelBuilding = true;
```
Comment (IvarWithoutBones) [inline pkgs/development/libraries/utf8cpp/default.nix:20]: This isn't required, nix figures this out automatically. From the nixpkgs manual:
`Unless set to false, some build systems with good support for parallel building including cmake, meson, and qmake will set it to true.`

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
a few minor nits, otherwise LGTM

**Comment by IvarWithoutBones** [comment]
Oh, 1 more thing: https://github.com/NixOS/nixpkgs/pull/89276/commits/2240aab436d96a47c0750e004d83a7db14409c36 and https://github.com/NixOS/nixpkgs/pull/89276/commits/5e82fc87b325ab2b7c2642992315eecd711beacc should be squashed.

**Comment by jobojeha** [comment]
Thanks for the review! :smile: Updated to your suggestions.

**Review by IvarWithoutBones — APPROVED** [review]
LGTM

**Review by danieldk — COMMENTED** [review]
Thanks for adding this derivation! Looks good to me :+1: , though I wondered why tests are disabled. They seem to work well and might catch future regressions.

Diff for enabling tests: https://gist.github.com/danieldk/88c0b97819b3c009aaee32be004a69a4

**Review by danieldk — APPROVED** [review]
Thanks for enabling the tests! LGTM.

Result of `nixpkgs-review pr 89276` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- utf8cpp
</details>

**Comment by jobojeha** [comment]
@danieldk Oh yes thats perfect. Thanks for your proposed changes! 😄  I pushed them now. I originally disabled tests because I looked into the build instructions over in the Arch PKGBUILD and with enabled tests it wasn't recognized by paraview as a dependency at first. I forgot about the doChecks, so I guess thats why.

**Pull request merged**